syncd/syncd_pfc_watchdog.cpp

```diff
@@ -0,0 +1,260 @@
+#include "syncd_pfc_watchdog.h"
+#include "syncd.h"
+
+#define PFC_WD_POLL_MSECS 50
+
+PfcWatchdog::PfcCounterIds::PfcCounterIds(
+        _In_ sai_object_id_t queue,
+        _In_ sai_object_id_t port,
+        _In_ const std::vector<sai_port_stat_t> &portIds,
+        _In_ const std::vector<sai_queue_stat_t> &queueIds):
+    queueId(queue), portId(port), portCounterIds(portIds), queueCounterIds(queueIds)
+{
+}
+
+void PfcWatchdog::setPortCounterList(
+        _In_ sai_object_id_t queueVid,
+        _In_ sai_object_id_t queueId,
+        _In_ const std::vector<sai_port_stat_t> &counterIds)
```

Comment:
when you query the port counter list, why do you give queueId? For example, for PFC TX and RX, you should be able to query for both queue 3 and queue 4 for the same port. You do not need to give queue id here.